lib/browser/api/dialog.js

```diff
@@ -70,70 +70,55 @@ const checkAppInitialized = function () {
   }
 }
 
-module.exports = {
-  showOpenDialog: function (...args) {
-    checkAppInitialized()
-
-    let [window, options, callback] = parseArgs(...args)
-
-    if (options == null) {
-      options = {
-        title: 'Open',
-        properties: ['openFile']
-      }
-    }
-
-    let { buttonLabel, defaultPath, filters, properties, title, message, securityScopedBookmarks = false } = options
-
-    if (properties == null) {
-      properties = ['openFile']
-    } else if (!Array.isArray(properties)) {
-      throw new TypeError('Properties must be an array')
-    }
-
-    let dialogProperties = 0
-    for (const prop in fileDialogProperties) {
-      if (properties.includes(prop)) {
-        dialogProperties |= fileDialogProperties[prop]
-      }
-    }
-
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
+const openDialog = (sync, window, options) => {
+  checkAppInitialized()
+
+  if (window.constructor !== BrowserWindow) options = window
+  if (options == null) {
+    options = {
+      title: 'Open',
+      properties: ['openFile']
     }
+  }
 
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
+  const {
+    buttonLabel = '',
+    defaultPath = '',
+    filters = [],
+    properties = ['openFile'],
+    title = '',
+    message = '',
+    securityScopedBookmarks = false
+  } = options
+
+  if (!Array.isArray(properties)) throw new TypeError('Properties must be an array')
+
+  let dialogProperties = 0
+  for (const prop in fileDialogProperties) {
+    if (properties.includes(prop)) {
+      dialogProperties |= fileDialogProperties[prop]
     }
+  }
 
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
+  if (typeof title !== 'string') throw new TypeError('Title must be a string')
+  if (typeof buttonLabel !== 'string') throw new TypeError('Button label must be a string')
+  if (typeof defaultPath !== 'string') throw new TypeError('Default path must be a string')
+  if (typeof message !== 'string') throw new TypeError('Message must be a string')
 
-    if (filters == null) {
-      filters = []
-    }
+  const settings = { title, buttonLabel, defaultPath, filters, message, securityScopedBookmarks, window }
+  settings.properties = dialogProperties
 
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
-      throw new TypeError('Message must be a string')
-    }
+  if (sync) return binding.showOpenDialogSync(settings)
+  binding.showOpenDialog(settings)
```

Comment:
missing return

Reply:
IMO this would be more readable
```
  if (sync) {
    return binding.showOpenDialogSync(settings)
  } else {
    return binding.showOpenDialog(settings)
  }
```